src/Microsoft.VisualStudio.Validation/Assumes.cs

```diff
@@ -59,7 +61,7 @@ public static void NotNullOrEmpty<T>([ValidatedNotNull, NotNull]ICollection<T> v
         /// </summary>
         /// <typeparam name="T">The type of value to test.</typeparam>
         [DebuggerStepThrough]
-        public static void NotNullOrEmpty<T>([ValidatedNotNull, NotNull]IEnumerable<T> values)
+        public static void NotNullOrEmpty<T>([ValidatedNotNull, NotNull]IEnumerable<T>? values)
```

Comment:
```suggestion
        public static void NotNullOrEmpty<T>([ValidatedNotNull, NotNull]IEnumerable<T?>? values)
```

Reply:
That produces a compiler error because `T : class` isn't a type constraint. Nulls are allowed already.